### PR TITLE
Adding service principal lock configuration property 

### PR DESCRIPTION
**Pull request by madansr7**
Schema PR: 6786198

### CODE REVIEW

**Comment by madansr7** [comment]
@FaithOmbongi can you please review and approve?

File: changelog/Microsoft.DirectoryServices.json [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [servicePrincipalLockConfiguration](https://docs.microsoft.com/en-us/graph/api/resources/servicePrincipalLockConfiguration?view=graph-rest-beta) resource",
+          "Target": "servicePrincipalLockConfiguration"
+        },
+        {
+          "Id": "7e11383c-72a9-465a-970d-471360f95418",
+          "ApiChange": "Property",
+          "ChangedApiName": "servicePrincipalLockConfiguration",
+          "ChangeType": "Addition",
+          "Description": "Added the [servicePrincipalLockConfiguration](https://docs.microsoft.com/en-us/graph/api/resources/application?view=graph-rest-beta) resource",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:18]: Tooling bug
```suggestion
          "Description": "Added the **servicePrincipalLockConfiguration** property to the [application](https://docs.microsoft.com/en-us/graph/api/resources/application?view=graph-rest-beta) resource.",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+          "Target": "application"
+        }
+      ],
+      "Id": "7e11383c-72a9-465a-970d-471360f95418",
+      "Cloud": "Prod",
+      "Version": "beta",
+      "CreatedDateTime": "2022-09-20T21:02:17.1897204Z",
+      "WorkloadArea": "Identity and access",
+      "SubArea": "Directory management"
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:27]:
```suggestion
      "WorkloadArea": "Applications",
      "SubArea": ""
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Hi @madansr7 - I've left some inline suggestions and aspects that need clarification.

Also, add the property to the [Update application](https://review.learn.microsoft.com/en-us/graph/api/application-update?view=graph-rest-beta&branch=pr-en-us-18517&tabs=http#request-body) doc as an updatable property.

File: api-reference/beta/resources/servicePrincipalLockConfiguration.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+author: "madansr7"
+---
+
+# servicePrincipalLockConfiguration resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Basic profile information of the application.
```
[the remark this answers is not in the document]
Comment (FaithOmbongi) [reply api-reference/beta/resources/servicePrincipalLockConfiguration.md:16]: @madansr7 - Please see the suggested update that matches this description with the description in line 3.
```suggestion
Specifies the sensitive properties of the app that are locked for editing on the service principal. You can disable this configuration to allow the properties to be updated.
```

File: api-reference/beta/resources/servicePrincipalLockConfiguration.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  "blockType": "resource",
+  "optionalProperties": [
+
+  ],
+  "@odata.type": "microsoft.graph.servicePrincipalLockConfiguration"
+}-->
+
+```json
+{
+   "isEnabled": "Boolean",
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/servicePrincipalLockConfiguration.md:41]: nit
```suggestion
   "@odata.type": "#microsoft.graph.servicePrincipalLockConfiguration",
   "isEnabled": "Boolean",
```

**Review by FaithOmbongi — APPROVED** [review]
Thank you @madansr7. I've left inline comments to clarify one question you asked and suggest changes for one oversight.

Otherwise, **approved with changes.**

PS: I notice that you copied an existing doc and used it as a template for the new resource type. I strongly recommend that you always use doc stubs to start authoring and avoid duplicating an existing doc. Doc stubs are always generated in your engineer's schema PR. Using the template helps ensure that you aren't inheriting any template "bugs" from existing docs.

**Pull request merged**